packages/trimerge-sync/src/TrimergeClient.ts

```diff
@@ -318,15 +328,33 @@ export class TrimergeClient<
     this.clientList = Array.from(this.clientMap.values());
     this.clientListSubs.emitChange(event);
   }
-  private sync(): void {
+  private async sync(): Promise<void> {
     const commits = this.unsyncedCommits;
     if (commits.length > 0 || this.newPresence !== undefined) {
       this.unsyncedCommits = [];
       this.updateSyncState({ localSave: 'saving' });
-      this.store.update(commits, this.newPresence);
+      this.numUnsavedCommits++;
+      try {
+        await this.store.update(commits, this.newPresence);
+        if (this.numUnsavedCommits > 0) {
```

Comment:
I would probably be conservative in this code and assert that it's > 0

Reply:
Not sure I understand. Do you want me to throw if it's not greater than 0?